src/Symfony/Component/DependencyInjection/Compiler/AutowirePass.php

```diff
@@ -115,50 +118,34 @@ private function completeDefinition($id, Definition $definition, $autowired)
     /**
      * Gets the list of methods to autowire.
      *
-     * @param \ReflectionClass     $reflectionClass
-     * @param true|string|string[] $autowired
+     * @param string           $id
+     * @param \ReflectionClass $reflectionClass
+     * @param string[]         $autowired
      *
      * @return \ReflectionMethod[]
      */
-    private function getMethodsToAutowire(\ReflectionClass $reflectionClass, $autowired)
+    private function getMethodsToAutowire($id, \ReflectionClass $reflectionClass, array $autowired)
     {
-        if (true === $autowired && $constructor = $reflectionClass->getConstructor()) {
-            return array($reflectionClass->getConstructor());
-        }
+        $found = array();
 
-        if (is_array($autowired)) {
-            return $this->getMethods($reflectionClass, $autowired);
-        }
+        foreach ($reflectionClass->getMethods(\ReflectionMethod::IS_PUBLIC) as $reflectionMethod) {
+            if ($reflectionMethod->isStatic()) {
+                continue;
+            }
 
-        return $this->getSetters($reflectionClass);
-    }
+            foreach ($autowired as $methodPattern) {
+                if ($methodPattern === $reflectionMethod->name || ('*' === substr($methodPattern, -1) && 0 === strpos($reflectionMethod->name, substr($methodPattern, 0, -1)))) {
```

Comment:
I suggest to turn the `$autowired` into a regexp at the beginning of the method, then using preg_match (in case insensitive mode) with e.g.:
`$autowiredRx = implode('|', array_map(function ($m) { return str_replace('\*', '.*', preg_quote($m, '/')); }, $autowired));`